packages/scenario/test/repeat.test.ts

```diff
@@ -2865,4 +2865,59 @@ describe('Tests ported from JavaRosa - repeats', () => {
 			);
 		});
 	});
+
+	describe('XFormParserTest.java', () => {
+		describe('form with `count-non-empty` func[tion]', () => {
+			interface NamespaceOptions {
+				readonly includeCommonNamespaces: boolean;
+			}
+
+			/**
+			 * **PORTING NOTES**
+			 *
+			 * - It's not clear if this belongs here. It's fundamentally a test of the
+			 *   XPath `count-non-empty` function, but appears to depend on repeat
+			 *   instances as part of exercising the functionality. Otherwise, it
```

Comment:
I think my question of where it belongs is really more around whether we should test XPath functionality in `scenario` in general, rather than whether it belongs in the repeat suite. I think there's probably a good case to be made for _some_ form-level integration test coverage of specific XPath behavior.

A bunch of the ported tests fit that category. I think the pattern I see where they feel most valuable in Web Forms' `scenario` suites is when they exercise the dependency graph.

In some other cases, I think there's been a lot of benefit in the hard boundary between the `xpath` and `xforms-engine` packages, and to the extent the XPath functionality can be tested outside XForms context I would probably err in that direction.

I'm not inclined to do anything on this one now, but I think that when it comes to my attention again I'd be inclined to make sure that there's equivalent coverage in `xpath`.